internal/supervisor/installation.go

```diff
@@ -116,13 +116,28 @@ func (s *InstallationSupervisor) Supervise(installation *model.Installation) {
 	}
 	defer lock.Unlock()
 
+	// Before working on the installation, it is crucial that we ensure that it
+	// was not updated to a new state by another provisioning server.
+	originalState := installation.State
+	installation, err := s.store.GetInstallation(installation.ID, true, false)
+	if err != nil {
+		logger.WithError(err).Errorf("Failed to get refreshed installation")
+		return
+	}
+	if installation.State != originalState {
+		logger.WithField("oldInstallationState", originalState).
+			WithField("newInstallationState", installation.State).
+			Warn("Another provisioner has worked on this installation; skipping...")
```

Comment:
Is another provisioner working on the installation something normal? What is the reason for warning?

Reply:
Yeah, it is. Multiple provisioners could end up working together to move any cloud objects from the beginning state to an end state.

That's how the `test` account and others are working now.

The warning is there to log that this stale state issue is happening in our accounts so I can verify that the fix is working. We may just silently exit in the future.